lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2232,7 +2232,7 @@ sub should_show_autoinst_log {
 sub should_show_investigation {
     my ($self) = @_;
 
-    return $self->state ne DONE || $self->result eq FAILED;
+    return ! grep { /^$self->state$/ } FINAL_STATES || grep { /^$self->result$/ } NOT_OK_RESULTS;
```

Comment:
I'd avoid calling `state`/`result` for every state/result we want to consider. And is is really required to turn it into a regex (`$_ eq $state` should work as well)? At least for the states one could also use `meta_state` and check whether it returns `FINAL`. I would also add braces to make sure the precedence is what we want.

Reply:
tried with

```perl
OpenQA::Jobs::Constants::meta_state($self->state) eq OpenQA::Jobs::Constants::FINAL || grep { /^$self->result$/ } NOT_OK_RE       SULTS;
```

Do you know why the previous code says "done || failed" and not "done && failed"?

Reply:
> Do you know why the previous code says "done || failed" and not "done && failed"?

No, I also think `&&` would make more sense.